src/jpeg/segment.rs

```diff
@@ -177,7 +177,14 @@ impl EncodeAt for JpegSegment {
             1 if !self.contents.is_empty() => Some(self.contents.clone()),
             2 if !self.entropy.is_empty() => Some(self.entropy.clone()),
             _ => {
-                *pos -= 1 + !self.contents.is_empty() as usize + !self.entropy.is_empty() as usize;
+                let decrement =
+                    1 + !self.contents.is_empty() as usize + !self.entropy.is_empty() as usize;
+                if *pos >= decrement {
+                    *pos -= decrement;
+                } else {
+                    // Prevent underflow
+                    *pos = 0;
+                }
```

Comment:
Should be more idiomatic

```suggestion
                // Prevent underflow with malformed JPEGs
                *pos = pos.saturating_sub(decrement);
```

Reply:
👍